### PR TITLE
Notifications and Reader: Fix Spotify shortcode render

### PR DESCRIPTION
**Pull request by ivan-ottinger**
#### Changes proposed in this Pull Request:
This commit addresses the issue where the Spotify shortcode isn't properly rendered in WPCOM notifications and Reader.

The proposed code checks whether it is running on WPCOM and then confirms the display context with the help of `/lib/display-context.php`.

Reader render: **before**
Markup_on_2021-08-18_at_14_06_31

Reader render: **after**
Markup_on_2021-08-18_at_14_10_25

Notification render: **before**
Markup_on_2021-08-18_at_14_08_00

Notification render: **after**
Markup_on_2021-08-18_at_14_09_36

#### Does this pull request change what data or activity we track or use?
No, it does not.

#### Testing instructions:
1. Apply the generated patch to WPCOM sandbox as it is outlined in p7H4VZ-30i-p2.
2. Make sure there is a subscriber on your site that can receive notifications on newly published posts.
Notification settings are located in the subscriber's account at `/following/manage` -> (gear icon) "Settings" -> "Notify me of new posts" (toggle):
Markup_on_2021-07-29_at_15_35_56
3. Log in to the account of the user that can publish posts on the site.
4. Create and publish a new post that has the `[spotify]` shortcode in it. Example shortcode that can be used: `[spotify https://open.spotify.com/playlist/5PKZSKuHP4d27SXO5fB9Wl]`.
5. Review the notification in the subscriber's account.
6. Review the post in the Reader.
7. The shortcode should be rendered correctly in both - notification and Reader - similar to the screenshots above.
8. The link in the notification should open in a new tab.

### CODE REVIEW

**Review by arcangelini — APPROVED** [review]
Tested this in WPCOM with the phab patch. Everything LGTM 🥇

File: projects/plugins/jetpack/modules/shortcodes/spotify.php [hunk 1]
```diff
@@ -48,6 +48,26 @@ function jetpack_spotify_shortcode( $atts = array(), $content = '' ) {
...
+				'<a href="%1$s" target="_blank" rel="noopener noreferrer">%1$s</a>',
+				esc_url( $id )
+			);
+		} elseif ( A8C\Display_Context\READER === $context ) {
+			return sprintf(
+				'<iframe src="%1$s" height="%2$s" width="%3$s"></iframe>',
+				esc_url( $embed_url ),
+				esc_attr( $atts['height'] ),
+				esc_attr( $atts['width'] )
+			);
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/shortcodes/spotify.php:67]: Is there a specific reason why we need that fallback, given that we render an iFrame below? Are the inline styles added to the iFrame below causing issues in the Reader?

Comment (ivan-ottinger) [reply projects/plugins/jetpack/modules/shortcodes/spotify.php:67]: > Is there a specific reason why we need that fallback, given that we render an iFrame below? Are the inline styles added to the iFrame below causing issues in the Reader?

Thanks for the comment Jeremy. Yes, that's right the fallback is there to fix the "black box" below the player due to stripped down inline styles in Reader.

Page itself:
Markup on 2021-09-10 at 15:45:26

Reader (without the fix):
Markup on 2021-09-10 at 15:46:09

That is the reason I suggested to use the `height` and `width` attributes directly.

I should have added clarification on that in the PR.

**Review by jeherve — APPROVED** [review]

**Comment by ivan-ottinger** [comment]
Successfully deployed to WordPress.com. Changeset ID: r231571-wpcom

**Pull request merged**